test/fail.js

```diff
@@ -0,0 +1,38 @@
+var assert = require('assert');
+
+var R = require('..');
+var eq = require('./shared/eq');
+
+
+describe('fail', function() {
+  it('catches failed promises', function(done) {
+    var p = Promise.reject(1);
+
+    var f = function(n) {
+      eq(n, 1);
+      done();
+    };
+
+    R.fail(f, p);
+  });
+
+  it('does nothing to successfully resolved promises', function(done) {
+    var asyncAddTwo = R.pipe(Promise.resolve.bind(Promise), R.then(R.inc), R.fail(R.inc), R.then(R.inc));
```

Comment:
What if the `fail` part was `R.fail(R.multiply(0))`?  That would make it more clear what does and doesn't run...